lib/intelligent_foods/api_client.rb

```diff
@@ -17,27 +17,41 @@ def authenticate!
       request = build_request_with_body(uri: uri, body: body)
       authorization = IntelligentFoods::Authorization::Basic.
                       factory(client_id: id, client_secret: secret)
-      response = execute_request(request: request, uri: uri,
+      response = execute_request(request: request,
                                  authorization: authorization)
       handle_authentication_response(response: response.data)
       self
     end
 
-    def execute_request(request:, uri:, authorization: default_authorization)
+    def post(path:, body:)
+      uri = URI(path)
+      request = Net::HTTP::Post.new(uri)
+      request.body = body.to_json
+      request["content-type"] = "application/json"
+      execute_request(request: request)
+    end
+
+    def delete(path:)
+      uri = URI(path)
+      request = Net::HTTP::Delete.new(uri)
+      execute_request(request: request)
+    end
+
+    def get(path:)
+      uri = URI(path)
+      request = Net::HTTP::Get.new(uri)
+      execute_request(request: request)
```

Comment:
example for my comment below this:
```suggestion
      execute_request(uri: request.uri, request: request)
```